src/main/java/uk/gov/hmcts/reform/prl/services/serviceofdocuments/ServiceOfDocumentsService.java

```diff
@@ -0,0 +1,787 @@
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ServiceOfDocumentsService {
+
+    public static final String GOV_NOTIFY_TEMPLATE = "govNotifyTemplate";
+    public static final String SEND_GRID_TEMPLATE = "sendGridTemplate";
+    public static final String SERVED_PARTY = "servedParty";
+    private final ObjectMapper objectMapper;
+    private final SendAndReplyService sendAndReplyService;
+    private final ServiceOfApplicationService serviceOfApplicationService;
+    private final ServiceOfApplicationEmailService serviceOfApplicationEmailService;
+    private final ServiceOfApplicationPostService serviceOfApplicationPostService;
+    private final DynamicMultiSelectListService dynamicMultiSelectListService;
+    private final UserService userService;
+    private final AllTabServiceImpl allTabService;
+    private final DocumentLanguageService documentLanguageService;
+    private final EmailService emailService;
+
+    @Value("${xui.url}")
+    private String manageCaseUrl;
+
+    @Value("${citizen.url}")
+    private String citizenDashboardUrl;
+
+    public Map<String, Object> handleAboutToStart(String authorisation,
+                                                  CallbackRequest callbackRequest) {
+
+        Map<String, Object> caseDataMap = new HashMap<>();
+        CaseData caseData = CaseUtils.getCaseData(callbackRequest.getCaseDetails(), objectMapper);
+
+        caseDataMap.put(CASE_TYPE_OF_APPLICATION, CaseUtils.getCaseTypeOfApplication(caseData));
+        caseDataMap.put(DISPLAY_LEGAL_REP_OPTION, CaseUtils.isCitizenCase(caseData) ? "No" : "Yes");
+        caseDataMap.put(
+            MISSING_ADDRESS_WARNING_TEXT,
+            serviceOfApplicationService.checkIfPostalAddressMissedForRespondentAndOtherParties(caseData)
+        );
+        caseDataMap.put("sodDocumentsList", List.of(element(DocumentsDynamicList.builder()
+                                                                .documentsList(sendAndReplyService.getCategoriesAndDocuments(
+                                                                    authorisation,
+                                                                    String.valueOf(caseData.getId())
+                                                                )).build())));
+        List<DynamicMultiselectListElement> otherPeopleList = dynamicMultiSelectListService.getOtherPeopleMultiSelectList(
+            caseData);
+        caseDataMap.put(SOA_OTHER_PARTIES, DynamicMultiSelectList.builder().listItems(otherPeopleList).build());
+        caseDataMap.put(
+            SOA_OTHER_PEOPLE_PRESENT_IN_CASE,
+            CollectionUtils.isNotEmpty(otherPeopleList) ? YesOrNo.Yes : YesOrNo.No
+        );
+        caseDataMap.put(SOA_RECIPIENT_OPTIONS, serviceOfApplicationService.getCombinedRecipients(caseData));
+        //ADD DATA CHECK TO HANDLE REPLACING EXISTING SOD DOCUMENTS PENDING MANAGER CHECK
```

Comment:
Is this comment about warning the user if there are already packs waiting for manager check?

Reply:
@shashisk97 Yes, I have added this to handle that scenario.
It has been added now as part of commit https://github.com/hmcts/prl-cos-api/pull/2758/commits/0a58508a036731331187dc7f9f455cf963566bd1